script.js

```diff
@@ -0,0 +1,250 @@
+      let [x, y] = generateRandomPosition();
+      while (!isBoxEmpty(x, y)) {
+        [x, y] = generateRandomPosition();
+      }
+      gameValues[x][y] = generateRandomInitialValue();
+    }
+  
+    console.log(gameValues);
+  };
+  
+  const setMatrixAfterMove = (direction) => {
+      const logDirection = {
+          right: "Right",
+          left: "Left",
+          up: "Up",
+          down: "Down"
+      };
+  
+      const updateRowCol = (arr) => {
+          let currNum = 0;
+          let newArr = [];
+  
+          arr.forEach(value => {
+              if (value !== 0) {
+                  if (currNum === 0) {
+                      currNum = value;
+                  } else if (currNum === value) {
+                      newArr.push(currNum * 2);
+                      score += (currNum * 2);
+                      currNum = 0;
+                  } else {
+                      newArr.push(currNum);
+                      currNum = value;
+                  }
+              }
+          });
+  
+          if (currNum !== 0) {
+              newArr.push(currNum);
+          }
+  
+          while (newArr.length < 4) {
+              newArr.unshift(0);
+          }
+  
+          return newArr;
+      };
+  
+      const transform = {
+          right: (row) => row.slice().reverse(),
+          left: (row) => row.slice(),
+          up: (col) => col.slice().reverse(),
+          down: (col) => col.slice()
+      };
+  
+      const updateGameValues = (isRow, index, newArr) => {
+          if (isRow) {
+              gameValues[index] = newArr;
+          } else {
+              for (let i = 0; i < 4; i++) {
+                  gameValues[i][index] = newArr[i];
+              }
+          }
+      };
+  
+      console.log(logDirection[direction]);
+  
+      const isRow = direction === "right" || direction === "left";
+      for (let i = 0; i < 4; i++) {
+          let arr = isRow ? gameValues[i] : gameValues.map(row => row[i]);
+          let transformedArr = transform[direction](arr);
+          let newArr = updateRowCol(transformedArr);
+          if (direction === "right" || direction === "up") newArr.reverse();
+          updateGameValues(isRow, i, newArr);
+      }
+  };
+  
+  
+  //The game will end when for each element in the 2D array, no element has the same value as its adjacent element. (Because if it did then we
+  //could combine them and the game would continue.
+  const isGameEnd = () => {
+      for (let row = 0; row < 4; row++) {
+          for (let col = 0; col < 4; col++) {
+              const value = gameValues[row][col];
+              if (
+                  (row > 0 && gameValues[row - 1][col] === value) ||
+                  (row < 3 && gameValues[row + 1][col] === value) ||
+                  (col > 0 && gameValues[row][col - 1] === value) ||
+                  (col < 3 && gameValues[row][col + 1] === value)
+              ) {
+                  return true;
+              }
+          }
+      }

```

Comment:
The return is a bit confusing. The isGameEnd should return true if there are no possible moves but I am returning a false. Realized it rn, will fix it later.